Mathlib/Algebra/Homology/HomotopyCategory/HomComplexCohomology.lean

```diff
@@ -0,0 +1,161 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+module
+
+public import Mathlib.Algebra.Homology.ShortComplex.Ab
+public import Mathlib.Algebra.Homology.HomotopyCategory.HomComplex
+public import Mathlib.Algebra.Homology.HomotopyCategory.Shift
+
+/-!
+# Cohomology of the hom complex
+
+Given `ℤ`-indexed cochain complexes `K` and `L`, and `n : ℤ`, we introduce
+a type `HomComplex.CohomologyClass K L n` which is the quotient
+of `HomComplex.Cocycle K L n` which identifies cohomologous cocycles.
+The reason for not applying the homology API to this complex is that
+`Cochain K L` can be considered both as a complex of abelian groups
+or as a complex of `R`-modules when the category is `R`-linear.
```

Comment:
That's unfortunate. Would it not work to take the existing homology API and beef it up with an `R`-module structure in the linear case?